linker/shared/src/test/scala/org/scalajs/linker/OptimizerTest.scala

```diff
@@ -168,6 +169,42 @@ class OptimizerTest {
     }
   }
 
+  @Test
+  def testOptimizerDoesNotEliminateRequiredStaticField_issue4021(): AsyncResult = await {
+    val StringType = ClassType(BoxedStringClass)
+    val fooGetter = m("foo", Nil, T)
+    val classDefs = Seq(
+        classDef(
+            MainTestClassName,
+            kind = ClassKind.Class,
+            superClass = Some(ObjectClass),
+            memberDefs = List(
+                trivialCtor(MainTestClassName),
+                // static var foo: java.lang.String
+                FieldDef(EMF.withNamespace(PublicStatic).withMutable(true),
+                    "foo", NON, StringType),
+                // static def foo(): java.lang.String = Test::foo
+                MethodDef(EMF.withNamespace(MemberNamespace.PublicStatic),
+                    fooGetter, NON, Nil, StringType, Some({
+                      SelectStatic(MainTestClassName, "foo")(StringType)
+                    }))(EOH, None),
+                // static def main(args: String[]) { println(Test::foo()) }
+                mainMethodDef({
+                  predefPrintln(ApplyStatic(EAF, MainTestClassName, fooGetter, Nil)(StringType))
+                })
+            )
+        )
+    )
+
+    for (linkingUnit <- linkToLinkingUnit(classDefs, MainTestModuleInitializers, TestIRRepo.fulllib)) yield {
```

Comment:
Do we need the `fullib` for this test? IIUC this test should work with the `minilib` (in other words: `println` is not required).

Reply:
I replaced by a `console.log` to get rid of the full lib. It's useful to send it to some black-hole like `console.log` so that the optimizer can never eliminate the whole thing as dead code.

Reply:
Ah, yes of course. I meant `println` specifically is not required (as opposed to `console.log` or something similar). I take it my comment was skewed from me looking at the classclass test, where we test the reachability chain of `println` specifically. In that sense `println` is "more" required there.